src/compiler/checker.ts

```diff
@@ -7173,6 +7173,25 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 }
             }
             else {
+                if(propertySymbol.flags & SymbolFlags.Accessor && propertySymbol.declarations && propertySymbol.declarations.length === 2) {
+                    const getter = isGetAccessor(propertySymbol.declarations[0]) ? propertySymbol.declarations[0] : isGetAccessor(propertySymbol.declarations[1]) ? propertySymbol.declarations[1] : undefined;
+                    const setter = isSetAccessor(propertySymbol.declarations[0]) ? propertySymbol.declarations[0] : isSetAccessor(propertySymbol.declarations[1]) ? propertySymbol.declarations[1] : undefined;
+                    if(getter && setter) {
+                        const getterReturnType = getter.type && getTypeFromTypeNode(getter.type);

```

Comment:
This shouldn't be syntactic - use `getTypeOfSymbol` and `getWriteTypeOfSymbol` to get the read and write types instead.